Automattic/wp-calypso
# A4A: Ensure updateSitesDashboardUrl only returns a URL, and add tests to confirm the results

Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/148

## Proposed Changes

* This PR moves the `page.replace` functionality outside of `updateSitesDashboardUrl`, so the function itself just returns a URL. In doing so, we can then easily test that the function returns the expected results.
* This PR adds tests for the `updateSitesDashboardUrl` function.

## Testing Instructions

To ensure the modified functionality in updateSitesDashboardUrl has not affected functionality:

* Run `yarn start-a8c-for-agencies` locally and then visit `http://agencies.localhost:3000/sites`.
* Attempt to add search, filters, click on favorites and open the preview pane - plus manually adding search / filter / preview pane modified URLs to make suer the experience is not broken.

To ensure the tests work:
* The tests in this PR should pass, but you can also run the tests with `yarn run test-client client/a8c-for-agencies/sections/sites/sites-dashboard/test`

## Pre-merge Checklist
